gp2gp-translator/src/integrationTest/java/uk/nhs/adaptors/pss/translator/EhrExtractHandlingIT.java

```diff
@@ -0,0 +1,171 @@
+package uk.nhs.adaptors.pss.translator;
+
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+
+import static com.jayway.awaitility.Awaitility.await;
+
+import static uk.nhs.adaptors.common.testutil.FileUtils.readResourceAsString;
+import static uk.nhs.adaptors.connector.model.MigrationStatus.EHR_EXTRACT_REQUEST_ACCEPTED;
+import static uk.nhs.adaptors.connector.model.MigrationStatus.EHR_EXTRACT_TRANSLATED;
+
+import java.util.List;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.json.JSONException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.skyscreamer.jsonassert.Customization;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+import org.skyscreamer.jsonassert.comparator.CustomComparator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.jms.core.JmsTemplate;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import lombok.SneakyThrows;
+import uk.nhs.adaptors.connector.dao.PatientMigrationRequestDao;
+import uk.nhs.adaptors.connector.service.MigrationStatusLogService;
+import uk.nhs.adaptors.pss.translator.mhs.model.InboundMessage;
+
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+@ExtendWith({SpringExtension.class})
+@DirtiesContext
+@AutoConfigureMockMvc
+public class EhrExtractHandlingIT {
+    private static final int NHS_NUMBER_MIN_MAX_LENGTH = 10;
+    private static final String EBXML_PART_PATH = "xml/ebxml_part.xml";
+    private static final String NHS_NUMBER_PLACEHOLDER = "{{nhsNumber}}";
+    private static final List<String> IGNORED_JSON_PATHS = List.of(
+        "id",
```

Comment:
I think this exclusion list is too big - most of IDs are taken from HL7, only few are random